### PR TITLE
docs: Module 3 Protobuf (B9lab updates 2-2)

### PR DESCRIPTION
**Pull request by coldice**
Docs content update for Module 3 (Main Concepts), Section _Protobuf_  - part of the [b9lab-content](https://github.com/cosmos/sdk-tutorials/tree/master/b9lab-content).

### Content state

**✔️ Completed**

### Technical review

- [x] requested
- [x] completed

### Language review

- [x] requested
- [x] completed

### CODE REVIEW

File: b9lab-content/3-main-concepts/09-protobuf.md [hunk 1]
```diff
@@ -1,9 +1,62 @@
 # Protobuf
 
+## Overview
+
+Protocol buffers, also called Protobuf in the short form, are an open source, extensible,  cross-platform and language-neutral method of serializing object data, primarily for network communication. Libraries for multiple platforms parse a common interface description language to generate source code for encoding and decoding streams of bytes representing structured data.
```
Comment (cmwaters) [inline b9lab-content/3-main-concepts/09-protobuf.md:5]:
```suggestion
Protocol buffers, also called Protobuf in the short form, are an open source, extensible, cross-platform and language-agnostic method of serializing object data, primarily for network communication and storage. Libraries for multiple languages parse a common interface description language to generate source code for encoding and decoding streams of bytes representing structured data. 
```

File: b9lab-content/3-main-concepts/09-protobuf.md [hunk 2]
```diff
@@ -1,9 +1,62 @@
...
+* List of **Store Keys**: Each module in the Cosmos SDK uses multistore to persist their part of the state. Access to such stores requires a list of keys that are declared in the type definition of the app.
+* List of each module’s **Keepers**: A Keeper is an abstract piece in each module to handle the module's interaction with stores, specify references to other modules' keepers, and implement other core functionalities of the module. For cross-module interactions to work, all modules in the Cosmos SDK need to have their keepers declared in the app's type definition and exported as interfaces to other modules so that the keeper's methods of one module can be called and accessed in other modules, when authorized.
+* Reference to **codec**: Defaulted to go-amino, the codec in your Cosmos SDK application can be substituted with other suitable encoding frameworks as long as they persist data stores in byte slices and are deterministic.
+Reference to Module Manager: A reference to an object containing a list of the applications modules, known as the Module Manager.
```
Comment (cmwaters) [inline b9lab-content/3-main-concepts/09-protobuf.md:54]:
```suggestion
* Reference to the Module Manager: This acts like a router to all the applications modules.
```

**Review by cmwaters — APPROVED** [review]
👍

File: b9lab-content/3-main-concepts/09-protobuf.md (hunk 2 above)
Comment (CitMC) [reply b9lab-content/3-main-concepts/09-protobuf.md:54]: @xavierlepretre

File: b9lab-content/3-main-concepts/09-protobuf.md [hunk 3]
```diff
@@ -1,62 +1,141 @@
...
+
+## Protobuf basics for Go
 
-## Protobuf Basics for Go
+The [gobs](https://golang.org/pkg/encoding/gob/) package for Go is a comprehensive package for the Go environment, but it doesn’t work well if you need to share information with applications written in other languages. Another challenge is how to contend with fields that may themselves contain information needing to be parsed or encoded.
 
-The [gobs](https://golang.org/pkg/encoding/gob/) package for Go is a comprehensive package for the Go environment but it doesn’t work well if you need to share information with applications written in other languages. Another challenge is how to contend with fields that may themselves contain information that needs to be parsed or encoded. 
+For example, a JSON or XML object may contain discrete fields that are stored in a string field. In another example, a time may be stored as two integers representing hours and minutes. A Protobuf encapsulated the necessary conversions in both directions. The generated classes provide getters and setters for the fields and take care of the details for reading and writing the message as a unit.
```
Comment (xavierlepretre) [inline b9lab-content/3-main-concepts/09-protobuf.md:33]:
```suggestion
For example, a JSON or XML object may contain discrete fields that are stored in a string field. In another example, a time may be stored as two integers representing hours and minutes. A Protobuf encapsulates the necessary conversions in both directions. The generated classes provide getters and setters for the fields and take care of the details for reading and writing the message as a unit.
```

**Pull request merged**